pkg/api/types.go

```diff
@@ -2154,3 +2154,32 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
+	// Who currently has the lock.
+	HeldBy string `json:"heldby"`
+	// How long the lock is valid.
+	LeaseTime uint64 `json:"duration"`
+	// When the lock was first acquired.
+	AcquiredTime string `json:"atime,omitempty"`
+	// When the lock was last renewed.
+	RenewTime string `json:"rtime,omitempty"`
+}
+
+// Lock represents the configuration of a resource lock.
+type Lock struct {
```

Comment:
Hi mike.  Its a good point you make, but there is a caveat. 

- The data structure doesn't enforce a lease, but.. 
-  `registry.go` does enforce one.It enforces the lease by actually setting the TTL in etcd itself.

So i think we could go either way here.  Slightly leaning towards lock since  leasing isn't necessarily enforced by this data structure.